### PR TITLE
[Feature branch: extensions] Adding initial support for extensions

### PR DESCRIPTION
**Pull request by saratvemulapalli**
### Description
 Adding initial support for extensions
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/transport/TransportService.java [hunk 1]
```diff
@@ -428,9 +428,9 @@ public void connectToNode(final DiscoveryNode node, ConnectionProfile connection
             // We don't validate cluster names to allow for CCS connections.
             handshake(newConnection, actualProfile.getHandshakeTimeout().millis(), cn -> true, ActionListener.map(listener, resp -> {
                 final DiscoveryNode remote = resp.discoveryNode;
-                if (node.equals(remote) == false) {
+                /*if (node.equals(remote) == false) {
```
Comment (owaiskazi19) [inline server/src/main/java/org/opensearch/transport/TransportService.java:431]: Can we add a TODO here and link the issue?

**Review by owaiskazi19 — APPROVED** [review]
Changes looks good to me. A small question, do we have an issue to figure out a way to read extension.yml of extensions in OpenSearch to know the plugin info?

**Comment by saratvemulapalli** [comment]
> ❌   Gradle Check failure [f92a8c2](https://github.com/opensearch-project/OpenSearch/commit/f92a8c23c0820a3169d24c9b04a6701c25b4464e) [Log 4259](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_4259.log)
> 
> [Reports 4259](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_4259_reports.zip)

```
 - org.opensearch.index.ShardIndexingPressureConcurrentExecutionTests.testReplicaThreadedUpdateToShardLimitsAndRejections
 - org.opensearch.transport.TransportServiceHandshakeTests.testNodeConnectWithDifferentNodeId
 - org.opensearch.transport.SniffConnectionStrategyTests.testConnectFailsIfNoConnectionsOpened
 ```

2,3 are due to issue: https://github.com/opensearch-project/OpenSearch/issues/2772
1 is a flaky test.

**Comment by saratvemulapalli** [comment]
> Changes looks good to me. A small question, do we have an issue to figure out a way to read extension.yml of extensions in OpenSearch to know the plugin info?

Sure: https://github.com/opensearch-project/OpenSearch/issues/1972
The changes are merged on my fork. I'll send out a PR to take care of this.

**Review by joshpalis — APPROVED** [review]
looks good to me :)

**Pull request merged**